src/rules/valid-expect.ts

```diff
@@ -48,6 +50,18 @@ const findPromiseCallExpressionNode = (node: TSESTree.Node) =>
     ? getPromiseCallExpressionNode(node.parent)
     : null;
 
+const findFirstAsyncFunction = ({
+  parent,
+}: TSESTree.Node): TSESTree.Node | null => {
+  if (!parent) {
+    return null;
+  }
+
+  return isFunction(parent) && parent.async
+    ? parent
+    : findFirstAsyncFunction(parent);
+};
```

Comment:
Sorry for the late response; I only saw this after the new release.

Doesn't this cause false-positives for non-async functions contained in async functions? It's probably a really rare situation in test files, but I think it would be more correct like this:

```ts
  if (!parent) {
    return null;
  }

  if (isFunction(parent)) {
    return parent.async ? parent : null;
  }

  return findFirstAsyncFunction(parent);
```

Reply:
Ah, I see the logic is changed in https://github.com/jest-community/eslint-plugin-jest/pull/1579 anyway.

Reply:
Do you have a test case that's failing? If so please share either way so we can ensure that going forward it doesn't fail

Reply:
No failing test case, just noticed this while reading the code.